components/script/dom/svgsvgelement.rs

```diff
@@ -5,34 +5,94 @@
 use crate::dom::attr::Attr;
 use crate::dom::bindings::codegen::Bindings::SVGSVGElementBinding;
 use crate::dom::bindings::inheritance::Castable;
-use crate::dom::bindings::root::{DomRoot, LayoutDom};
+use crate::dom::bindings::reflector::DomObject;
+use crate::dom::bindings::root::{DomRoot, LayoutDom, Dom};
 use crate::dom::bindings::str::DOMString;
 use crate::dom::document::Document;
 use crate::dom::element::{AttributeMutation, Element, RawLayoutElementHelpers};
-use crate::dom::node::Node;
+use crate::dom::node::{Node, window_from_node};
 use crate::dom::svggraphicselement::SVGGraphicsElement;
 use crate::dom::virtualmethods::VirtualMethods;
 use dom_struct::dom_struct;
 use html5ever::{LocalName, Prefix};
 use script_layout_interface::SVGSVGData;
 use style::attr::AttrValue;
+use canvas_traits::webgl::{WebGLMsg, WebGLContextShareMode, WebGLMsgSender, webgl_channel, WebGLVersion, GLContextAttributes};
+use std::cell::{Cell, RefCell, Ref};
+use euclid::Size2D;
+use crate::dom::window::Window;
+use crate::dom::bindings::cell::DomRefCell;
+use std::borrow::Borrow;
+use crate::dom::webglcontextevent::WebGLContextEvent;
+use crate::dom::event::{EventBubbles, EventCancelable, Event};
 
 const DEFAULT_WIDTH: u32 = 300;
 const DEFAULT_HEIGHT: u32 = 150;
 
 #[dom_struct]
 pub struct SVGSVGElement {
     svggraphicselement: SVGGraphicsElement,
+    #[ignore_malloc_size_of = "Channels are hard"]
+    webgl_sender: Option<WebGLMsgSender>,
+    share_mode: WebGLContextShareMode,
+    #[ignore_malloc_size_of = "Defined in webrender"]
+    webrender_image: Cell<Option<webrender_api::ImageKey>>
 }
 
 impl SVGSVGElement {
+    #[allow(unrooted_must_root)]
     fn new_inherited(
         local_name: LocalName,
         prefix: Option<Prefix>,
         document: &Document,
     ) -> SVGSVGElement {
-        SVGSVGElement {
-            svggraphicselement: SVGGraphicsElement::new_inherited(local_name, prefix, document),
+        let window = document.window();
+
+        let webgl_chan = match window.webgl_chan() {
+            Some(chan) => chan,
+            None => panic!("Crash the system!"),
+        };
+
+        let (sender, receiver) = webgl_channel()
+            .unwrap();
+        let attrs = GLContextAttributes{
+            depth: false,
+            stencil: false,
+            alpha: true,
+            antialias: true,
+            premultiplied_alpha: true,
+            preserve_drawing_buffer: false,
+        };
+        let size = Size2D::new(DEFAULT_WIDTH, DEFAULT_HEIGHT);
+        webgl_chan
+            .send(WebGLMsg::CreateContext(WebGLVersion::WebGL1, size, attrs, sender))
```

Comment:
So, in new_inherited how would I go about getting the size for WebGLMsg::CreateContext? If I use the technique used in data later on then I would be using unsafe functions which I imagine wouldn't be okay inside new_inherited.

Reply:
Great question! We can issue a canvas resize command from layout to work around this. Just create a 1x1 context in the constructor for now.